jedi_language_server/jedi_utils.py

```diff
@@ -68,20 +71,23 @@ def wrapper(func: Callable[P, None]) -> Callable[P, None]:
 
         @functools.wraps(func)
         def debounced(*args: P.args, **kwargs: P.kwargs) -> None:
+            _debounce_semaphore.acquire()
+
             sig = inspect.signature(func)
             call_args = sig.bind(*args, **kwargs)
             key = call_args.arguments[keyed_by] if keyed_by else None
 
             def run() -> None:
                 with lock:
                     del timers[key]
-                return func(*args, **kwargs)
+                func(*args, **kwargs)
+                _debounce_semaphore.release()
```

Comment:
Any reason we're not wrapping this in a try/finally? Haven't fully wrapped my head around the implication, but it feels like we might want to release the semaphore even if an exception occurs in `func`.

Reply:
Good catch!